orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -n "$TARGET_JAVA_VERSION" ]; then
+  CURRENT_JAVA_VERSION=$(java -version 2>&1 | grep -oP "version [^0-9]?(1\.)?\K\d+" || true)
+
+  if [ -z "$CURRENT_JAVA_VERSION" ]; then
+    echo "Failed to determine the current java version."
+    exit 1;
+  elif [ $CURRENT_JAVA_VERSION -eq $TARGET_JAVA_VERSION ]; then
+    echo "Current java version equals to the target version."
+    exit 0;
+  elif [ $CURRENT_JAVA_VERSION -eq 8 ] && [ $TARGET_JAVA_VERSION -eq 11 ]; then
+    echo "Change java version from $CURRENT_JAVA_VERSION to $TARGET_JAVA_VERSION"
+
+    alternatives --set java java-11-openjdk.x86_64
+    ln -sfn /etc/alternatives/java_sdk_11 /usr/lib/jvm/java
+    mkdir -p /etc/alternatives/java_sdk_11/jre/lib/security
+    ln -sfn /etc/alternatives/java_sdk_11/conf/security/java.security /etc/alternatives/java_sdk_11/jre/lib/security/java.security
+    ln -sfn /etc/pki/java/cacerts /etc/alternatives/java_sdk_11/jre/lib/security/cacerts
+
+    #Regenerate the resolv.conf
+    source /etc/dhcp/dhclient-enter-hooks
+    generate_resolv_conf
```

Comment:
why is this required?

Reply:
why is this required to run at this point, when the image already contains the right script?

Reply:
Because the script runs before the orchestration and later as well triggered by the OS. We need the right content when starting CM so we can wait for the OS to run it.

Reply:
so we already have this state: https://github.com/hortonworks/cloudbreak/pull/14247/files#diff-a628a2b611d334912be2f1c73176d08e5ff1123201748b7dbcaf9d207c9a36f5R155
this could be used, preferably with file.replace and with append option, so we could avoid adding this dependency to this script. so there would be only 2 places where this domain thing is touched, instead of 3